packages/playground/website/src/components/missing-site-modal/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { Button, Flex, FlexItem } from '@wordpress/components';
+import { Modal } from '../modal';
+import { SitePersistButton } from '../site-manager/site-persist-button';
+import {
+	useAppDispatch,
+	useAppSelector,
+	selectActiveSite,
+} from '../../lib/state/redux/store';
+import { setActiveModal } from '../../lib/state/redux/slice-ui';
+import { selectClientInfoBySiteSlug } from '../../lib/state/redux/slice-clients';
+
+export function MissingSiteModal() {
+	const dispatch = useAppDispatch();
+	const closeModal = () => dispatch(setActiveModal(null));
+
+	const activeSite = useAppSelector((state) => selectActiveSite(state));
+	const clientInfo = useAppSelector(
+		(state) =>
+			activeSite?.slug &&
+			selectClientInfoBySiteSlug(state, activeSite?.slug)
+	);
+
+	if (!activeSite) {
+		return null;
+	}
+	if (activeSite.metadata.storage !== 'none') {
+		return null;
+	}
+
+	// TODO: Improve language for this modal
+	return (
+		<Modal
+			title="Save to browser storage?"
+			contentLabel="This is a dialog window which overlays the main content of the
+				page. It offers the user a choice between using a temporary Playground
+				and a persistent Playground that is saved to browser storage."
+			isDismissible={false}
+			shouldCloseOnClickOutside={false}
+			onRequestClose={closeModal}
+		>
+			<p>
+				WordPress Playground tried to load a Playground that does not
+				exist, so we loaded a temporary Playground instead. Any changes
+				to temporary Playgrounds are lost on page refresh.
+			</p>
+			<p>Would you like to save this Playground to browser storage?</p>
```

Comment:
Would something like this work better?

```suggestion
			<p>
				The {activeSite.metadata.name} Playground does not exist, so we
				loaded a temporary Playground instead.
			</p>
			<p>
				If you want to preserve your changes, you can save the
				Playground to browser storage.
			</p>
```